### PR TITLE
Migrate podgc controller to contextual logging

### PR DESCRIPTION
**Pull request by pravarag**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
With this code change, we are migrating podgc controller to contextual logging
#### Which issue(s) this PR fixes:

Fixes #
Part of https://github.com/kubernetes/enhancements/issues/3077
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sftim** [comment]
I think this does need a release note, as end users could observe this change.

File: pkg/controller/podgc/gc_controller.go [hunk 1]
```diff
@@ -97,11 +97,13 @@ func NewPodGCInternal(ctx context.Context, kubeClient clientset.Interface, podIn
 }
 
 func (gcc *PodGCController) Run(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+
 	defer utilruntime.HandleCrash()
 
-	klog.Infof("Starting GC controller")
+	logger.Info("Starting GC controller")
```
Comment (Namanl2001) [inline pkg/controller/podgc/gc_controller.go:104]:
```suggestion
	klog.FromContext(ctx).Info("Starting GC controller")
```
no need for an extra variable

File: pkg/controller/podgc/gc_controller.go [hunk 2]
```diff
@@ -112,19 +114,19 @@ func (gcc *PodGCController) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
-func (gcc *PodGCController) gc(ctx context.Context) {
+func (gcc *PodGCController) gc(logger klog.Logger, ctx context.Context) {
```
Comment (Namanl2001) [inline pkg/controller/podgc/gc_controller.go:117]: I believe logger can be retrived from context `logger := klog.FromContext(ctx)`
not sure why you've taken `logger` as input, similar for other funcs

**Review by Namanl2001 — COMMENTED** [review]
Thanks for the contribution. Probably you can see the things I pointed out in your IDE or in the failing jobs log: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/114689/pull-kubernetes-e2e-kind/1607580766642900992

**Comment by pohly** [comment]
/wg structured-logging

@pravarag: ping - do you intend to finish this PR?

**Comment by pravarag** [comment]
> /wg structured-logging
> 
> @pravarag: ping - do you intend to finish this PR?

Hey sorry, been caught up in few other tasks I missed this one. I'll work on finishing it soon now :)

**Review by pohly — COMMENTED** [review]
Please also remove the exception from https://github.com/kubernetes/kubernetes/blob/5f23f835879b514e9bf16809059b38957d20a486/hack/logcheck.conf#L51

**Comment by mengjiao-liu** [comment]
/sig instrumentation

Are you still working on this?

**Comment by pravarag** [comment]
> /sig instrumentation
> 
> Are you still working on this?

Hi, I've been working on fixing this since the past week. Will push the changes soon

**Comment by pohly** [comment]
@pravarag: myself and the approver for these kind of PRs are only available two more days. If you can't finish the PR now, please say so and let someone else finish it.

**Comment by pravarag** [comment]
> @pravarag: myself and the approver for these kind of PRs are only available two more days. If you can't finish the PR now, please say so and let someone else finish it.

Hi @pohly, apologies for delaying work on this PR. As I'm stuck in some other work, happy for anyone else to finish this. Kindly assign someone else who might be interested.

**Comment by pohly** [comment]
/close

Replaced by https://github.com/kubernetes/kubernetes/pull/119250

**Pull request closed without merging**